### PR TITLE
Setup New Document Table

### PR DESCRIPTION
**Pull request by djnunez-aot**
-Add Document Table for management plan, supporting documents and consultation record

-I think we should keep the upload document drag and drop, alot of users might use the drag and drop to upload files going to talk to MA

-Change Submit Management Plan button to Submit to EAO (https://eao-dst.atlassian.net/browse/SUBMIT-150)

### CODE REVIEW

File: submit-web/src/components/DocumentUpload/PendingDocumentRow.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+export default function PendingDocumentRow({
+  documentItem,
+  error = false,
+}: DocumentTableRowProps) {
+  const { name } = documentItem;
+
+  const onActionClick = () => {};
+
+  return (
+    <>
```
Comment (jadmnsaad) [inline submit-web/src/components/DocumentUpload/PendingDocumentRow.tsx]: what do you need this fragment for?

File: submit-web/src/components/DocumentUpload/DocumentTableRow.tsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+              cursor: "pointer",
+              mx: 0.5,
+              textDecoration: "none",
+            }}
+          >
+            <MuiLink onClick={getObjectFromS3} sx={{ textDecoration: "none" }}>
+              {name}
+            </MuiLink>
+          </Typography>
+        </StyledTableCell>
```
Comment (jadmnsaad) [inline submit-web/src/components/DocumentUpload/DocumentTableRow.tsx]: Is there a better naming more related to the document table than StyledTableCell

File: submit-web/src/components/DocumentUpload/DocumentTableRow.tsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+      link.click();
+    } catch (e) {
+      notify.error("Failed to download documentItem");
+    } finally {
+      setPendingGetObject(false);
+    }
+  };
+
+  return (
+    <>
```
Comment (jadmnsaad) [inline submit-web/src/components/DocumentUpload/DocumentTableRow.tsx]: what do you need this fragment for?

File: submit-web/src/components/DocumentUpload/PendingDocumentRow.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import { PackageTableRow, StyledTableCell } from "./DocumentTableRow";
+
+type DocumentTableRowProps = {
+  documentItem: {
+    id: number;
+    name: string;
+  };
+  error?: boolean;
+};
+export default function PendingDocumentRow({
```
Comment (jadmnsaad) [inline submit-web/src/components/DocumentUpload/PendingDocumentRow.tsx:19]: how is the document being uploaded here? I don't see a save object to s3 call

Comment (djnunez-aot) [reply submit-web/src/components/DocumentUpload/PendingDocumentRow.tsx:19]: fixing that right now all the other comments have been adressed, it was working before i added this

Comment (djnunez-aot) [reply submit-web/src/components/DocumentUpload/PendingDocumentRow.tsx:19]: fixed

File: submit-web/src/components/DocumentUpload/PendingDocumentRow.tsx (hunk 1 above)
Comment (djnunez-aot) [reply submit-web/src/components/DocumentUpload/PendingDocumentRow.tsx]: removed

File: submit-web/src/components/DocumentUpload/DocumentTableRow.tsx (hunk 2 above)
Comment (djnunez-aot) [reply submit-web/src/components/DocumentUpload/DocumentTableRow.tsx]: renamed to documenttablecell

File: submit-web/src/components/DocumentUpload/DocumentTableRow.tsx (hunk 3 above)
Comment (djnunez-aot) [reply submit-web/src/components/DocumentUpload/DocumentTableRow.tsx]: removed

**Review by jadmnsaad — APPROVED** [review]

**Pull request merged**